### PR TITLE
Refactored to use FlatList in place of deprecated ListView

### PR DESCRIPTION
**Pull request by ryanphung**
This is an improved version of the pull request created by @rollymaduk at #14. Changes:

- Removed the `.idea` folder
- Fixed an issue where the scope of `this.renderList()` function is gone due to its assignment in the expression `(renderList || this.renderList)`
- Resolve conflicts in `index.js`
- Adding SafeAreaView to KeyboardAvoidingView - otherwise on iPhone X the textbox will be pushed up too high and covered by the "notch"

This PR addresses issue #8, and probably replace PR #14 and PR #38.

@hiddentao do let me know if any updates is needed. Looking forward to having it merged because otherwise the component breaks in RN 0.60. Thank you!

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,55 +1,61 @@
...
 } from 'react-native'
 
-
-import styles from './styles'
-
+import styles from "./styles"
 
 export default class ModalFilterPicker extends Component {
-  constructor (props, ctx) {
+  static keyExtractor(item, index) {
```
Comment (hiddentao) [inline src/index.js]: Does referencing this static method via `this.keyExtractor` work?

**Comment by hiddentao** [comment]
Great work! I see you're still adding commits though. I shall wait until you feel it's complete.

File: src/index.js (hunk 1 above)
Comment (ryanphung) [reply src/index.js]: @hiddentao This should be resolved by using `this.constructor.keyExtrator` in the latest commit (b316486).

**Comment by ryanphung** [comment]
@hiddentao it should be completed now. Let me know if you identify anything that needs to be changed or fixed!

**Comment by ryanphung** [comment]
@hiddentao and thank you for reviewing!

File: src/index.js [hunk 2]
```diff
@@ -113,68 +124,68 @@ export default class ModalFilterPicker extends Component {
...
-          {...listViewProps}
-          dataSource={ds}
-          renderRow={this.renderOption}
-          keyboardShouldPersistTaps={keyboardShouldPersistTaps}
-        />
-      )
     }
+    return (
+      <FlatList
+        keyExtractor={keyExtractor || this.keyExtractor}
```
Comment (hiddentao) [inline src/index.js]: Needs to be `this.constructor.keyExtractor`

**Comment by hiddentao** [comment]
@ryanphung Looks good, just one small fix left.

File: src/index.js (hunk 2 above)
Comment (ryanphung) [reply src/index.js]: Ah! Thanks for pointing out. It's fixed now @hiddentao

**Comment by ryanphung** [comment]
@hiddentao Fixed in latest commits.

**Comment by hiddentao** [comment]
Thanks!

**Comment by nihp** [comment]
Whether it is working fine now?

**Pull request merged**